web-client/public/js/setup-handlers.js

```diff
@@ -34,4 +34,9 @@ export const setupHandlers = grnState => {
         grnState.normalizationMax = $("#edge-weight-normalization-factor-menu").val();
         updateApp(grnState);
     });
+
+    if (!$("#colorEdges").hasClass("active")) {
+        grnState.colorOptimal = false;
+        updateApp(grnState);
+    }
```

Comment:
This doesn't look like a handler. `$("#colorEdges").hasClass("active")` is _not_ an event like `click` or `change`. In that light, this is a backwards move: it changes `grnState` based on the state of the app. Remember, we want the opposite: `grnState` _dictates_ the state of the app.

`setupHandlers` changes `grnState` _when the user performs an action_. There is no action here.